csrc/ir_nodes.cpp

```diff
@@ -2490,6 +2490,32 @@ IterDomain* IterDomain::resize(
 // vectorize to the left of the computeAt domain, and could allow us to do some
 // simple validation of vectorize as it's inputs are right most and contiguous.
 void IterDomain::parallelize(ParallelType t) {
+  // assert check that we only parallelize a leaf domain.
+  // Ideally, leaf domains are domains that are not used by any other domains.
+  // However, this is not true if reduandant split/merge operations are added
+  // by the user manually. e.g. case FusionAdvancedComputeAt8_CUDA.
+  // so the check is relaxed to include intermediate domains between root and
+  // leaf.
+  auto isLeafOrIntermediateDomain = [&]() {
+    if (definition()) {
```

Comment:
Why is this? Any non-root is supposed to be allowed? Then, why is the below check?

Reply:
below check is to make sure if a root-domain is not being used then it is also a leaf domain.

Reply:
I'm not sure what you mean. This returns true for any domain that has a definition. Do you have an example domain?